src/Migrations.php

```diff
@@ -537,7 +537,7 @@ private static function connectionSetup(array $options): void
             if (empty($options['directory'])) {
                 $path = defined('BASE_PATH') ?
                     constant('BASE_PATH') :
-                    defined('APP_PATH') ? dirname(constant('APP_PATH')) : '';
+                    (defined('APP_PATH') ? dirname(constant('APP_PATH')) : '');
```

Comment:
Could we move all this functionality to a dedicated (private) method to make  it easier to read and to reduce [cyclomatic complexity ](https://en.wikipedia.org/wiki/Cyclomatic_complexity).